pkg/proxy/endpoints.go

```diff
@@ -72,10 +78,11 @@ func (info *BaseEndpointInfo) Equal(other Endpoint) bool {
 	return info.String() == other.String() && info.GetIsLocal() == other.GetIsLocal()
 }
 
-func newBaseEndpointInfo(IP string, port int, isLocal bool) *BaseEndpointInfo {
+func newBaseEndpointInfo(IP string, port int, isLocal bool, topology map[string]string) *BaseEndpointInfo {
```

Comment:
Do we allow nil for `topology` here or should we interpret `nil` as empty map?

Reply:
`topology` map may only be used if both `EndpintSlice` and `ServiceTopology` are enabled, so `topology` will not be nil if it's been used (used only in `FilterTopologyEndpoint` function). For safety, I think `topology` map can be nil, but if it is nil when it was been used, just "continue". I'm gonna add a "continue" to `FilterTopologyEndpoint`.